### DISCUSSION TITLE
Forecasting and tuning Neural Prophet

### DISCUSSION THREAD

**Discussion by Msaleh87** [ourownstory/neural_prophet#741] [Q&A - forecasting best practices] [answered]
It was a great help just telling me that the retraining was disabled, it saved me a lot of searching time, thanks!

So I have an electric load data for about a year, I trained the model for a data set from Mar 2021 until Mar 2022. I enabled the holidays. Here is my code:

```
m = NeuralProphet(n_forecasts=(2*24*12),
    n_lags=(16*24*12),
    yearly_seasonality="auto",
    weekly_seasonality="auto",  
    daily_seasonality="auto", 
    growth="off", seasonality_reg= 0.5, optimizer = 'AdamW',
    learning_rate = 0.1, n_changepoints="auto", changepoints_range = "auto" ) # trend_reg=1
m.add_country_holidays("US", mode="additive") # lower_window=-1, upper_window=1
metrics = m.fit(L12M, freq="5min")

m.highlight_nth_step_ahead_of_each_forecast(step_number=m.n_forecasts)

P_horizon = 576 # 2 days ahead
future = m.make_future_dataframe(df = L12M, periods = P_horizon, n_historic_predictions=576)
forecast1 = m.predict(df=future)
```
That's the 48 hours ahead I got, I was wondering what hyperparamters I could add/tune to have better results. Or this is as good as it gets since I am using 1 feature (holidays)?

![Figure 2 (4)](https://user-images.githubusercontent.com/42920113/186675589-f5f8a7d0-3da7-4156-9182-6da9b478a299.png)

**Comment by ourownstory** [comment, marked as answer]
Hi @Msaleh87 Here are some Ideas:
* Try a lower amount of lags (e.g. just 3-7 days). OR downsample your data to 15min or hourly OR add a hidden layer of a small size (e.g. 8-32) (reason: your current AR model is massive, it has 2\*24\*12 \* 16\*24\*12 = 2,654,208 parameters, and is likely overfitting)
* instead of regularizing seasonality, try setting the number of fourier terms (instead of "auto")
* use the plot_components and plot_parameters function and see if the make sense
* find external regressors like temp
* at the end, tune learning rate and epochs

**Reply by ourownstory** [reply]
as a first step, try all seasons auto, except daily, set that to ~8
second, try subsampling to 15min
third, try keeping your lags and forecast settings, but adding a single hidden layer of size 8-32, and find a good learning rate.

**Reply by Msaleh87** [reply]
Thanks for getting back to me,
Why 8? I thought it should've multiple of 6 to have the sine and cosine terns of the Fourier series.
Also, when u say, try keeping lags and forecast settings, what do u mean?
How can I add single hidden layer, just put it in the parameters to be equal to 1, and rhe number of neurons to 8-32, is that what u mean?

**Reply by Msaleh87** [reply]
I will try these parameters and post the results here
